### PR TITLE
Enable String.Replace disabled tests

### LINKED ISSUE #18459: New string.Replace overloads OrdinalIgnoreCase comparison is broken.
```cs
[Fact]
public void Replace_IgnoreCase()
{
    // Nope: "abc"
    Assert.Equal("def", "abc".Replace("ABC", "def", StringComparison.OrdinalIgnoreCase));

    // Fine: "def"
    Assert.Equal("def", "abc".Replace("ABC", "def", StringComparison.InvariantCultureIgnoreCase));
    Assert.Equal("def", "abc".Replace("ABC", "def", StringComparison.CurrentCultureIgnoreCase));
}

[Fact]
public void EndsWithHyphen()
{
    // Nope: "def"
    Assert.Equal("abc", "abc".Replace("abc\u00AD", "def", StringComparison.OrdinalIgnoreCase));
}
```

Extracted from #1271. I haven't yet got a fix, I'm waiting for @tarekgh to check out the second issue

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes #18459

### CODE REVIEW

**Comment by tarekgh** [comment]
https://github.com/dotnet/coreclr/pull/11039

**Comment by tarekgh** [comment]
CC @hughbe @stephentoub @danmosemsft

File: src/System.Runtime/tests/System/StringTests.netcoreapp.cs [hunk 1]
```diff
@@ -264,7 +262,6 @@ public void Replace_StringComparsion_NullOldValue_ThrowsArgumentException()
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/coreclr/pull/11001")]
         public void Replace_StringComparsion_EmptyOldValue_ThrowsArgumentException()
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/StringTests.netcoreapp.cs]: typo "comparsion"

Comment (danmosemsft) [inline src/System.Runtime/tests/System/StringTests.netcoreapp.cs]: in 2 places in this file

**Comment by tarekgh** [comment]
> Have we got sufficient coverage of locale-aware replace? Someone mentioned replacing ß with ss -- I don't know if that's valid, but that kind of thing.

This PR is just enabling the tests which match the fixes we did related to ordinal ignore case and the recursion problem that @hughbe fixed. Also I am seeing the current tests covering some linguistic cases (e.g. Turkish I cases). We still can consider adding more linguistic cases as you mentioned. @hughbe do you want help with that?

**Comment by hughbe** [comment]
I did the Turkish I that tests current/invariant/ordinal differences.
I can take a look at the German(?) case you mentioned
Also ill see if there are any situations where ordinal and invariant give different results

**Comment by tarekgh** [comment]
> Also ill see if there are any situations where ordinal and invariant give different results

Invariant and ordinal can give different results. Invariant still linguistic operation (which is close to en-US behavior). ordinal is just ordinal.

**Comment by hughbe** [comment]
> Invariant and ordinal can give different results. Invariant still linguistic operation (which is close to en-US behavior). ordinal is just ordinal.

That's what I mean - I don't think we have much coverage for that right now

**Comment by danmosemsft** [comment]
Yay @hughbe you can put your your resume (albeit small and at the bottom most likely) that you own a method on String...

**Pull request merged**